src/components/shapes/attributes.js

```diff
@@ -224,6 +225,52 @@ module.exports = templatedArray('shape', {
             '`config.editable` or `config.edits.shapePosition`.'
         ].join(' ')
     },
-
+    label: {
+        text: {
+            valType: 'string',
+            dflt: '',
+            editType: 'arraydraw',
+            description: 'Sets the text to display with shape.'
+        },
+        font: fontAttrs({
+            editType: 'calc+arraydraw',
+            colorEditType: 'arraydraw',
+            description: 'Sets the shape label text font.'
+        }),
+        textposition: {
+            valType: 'enumerated',
+            values: [
+                'top left', 'top center', 'top right',
+                'middle left', 'middle center', 'middle right',
+                'bottom left', 'bottom center', 'bottom right',
+                'start', 'middle', 'end',
+            ],
+            editType: 'arraydraw',
+            description: 'Sets the position of the label text relative to the shape.'
+        },
+        textangle: {
+            valType: 'angle',
+            dflt: 0,
+            editType: 'calc+arraydraw',
+            description: [
+                'Sets the angle at which the `text` is drawn',
+                'with respect to the horizontal.'
+            ].join(' ')
+        },
+        xanchor: extendFlat({}, annAttrs.xanchor, {
+            description: 'Sets the text box\'s horizontal position anchor.',
+        }),
+        yanchor: extendFlat({}, annAttrs.yanchor, {
+            description: 'Sets the text box\'s vertical position anchor.',
+        }),
+        padding: {
+            valType: 'number',
+            dflt: 3,
+            editType: 'arraydraw',
+            description: 'Sets padding between edge of label and xanchor / yanchor.'
+        },
+        dflt: null,
```

Comment:
what's this `dflt`? 🔪 ?

Reply:
@alexcjohnson Good catch. This is associated with the top-level `label` property. Is it ok to have no `dflt` value here?

Reply:
We never `coerce('label')` so this `dflt` won't do anything.